erpnext/payroll/doctype/overtime_slip/overtime_slip.py

```diff
@@ -0,0 +1,186 @@
+		records = []
+		if overtime_based_on == "Attendance":
+			records = self.get_attendance_record()
+			if len(records):
+				self.create_overtime_details_row_for_attendance(records)
+		elif overtime_based_on == "Timesheet":
+			records = self.get_timesheet_record()
+			if len(records):
+				self.create_overtime_details_row_for_timesheet(records)
+		else:
+			frappe.throw(_('Select "Calculate Overtime Hours Based On" in Payroll Settings'))
+
+		if len(self.overtime_details):
+			self.total_overtime_duration = sum([int(detail.overtime_duration) for detail in self.overtime_details])
+
+		if not len(records):
+			self.overtime_details = []
+			frappe.msgprint(_("No {0} records found for Overtime").format(overtime_based_on))
+
+	def create_overtime_details_row_for_attendance(self, records):
+		self.overtime_details = []
+		for record in records:
+			if record.standard_working_time:
+				standard_working_time = record.standard_working_time
+			else:
+				standard_working_time = frappe.db.get_single_value("HR Settings", "standard_working_hours") * 3600
+				if not standard_working_time:
+					frappe.throw(_('Please Set "Standard Working Hours" in HR settings'))
+
+			if record.overtime_duration:
+				self.append("overtime_details", {
+					"reference_document_type": "Attendance",
+					"reference_document": record.name,
+					"date": record.attendance_date,
+					"overtime_type": record.overtime_type,
+					"overtime_duration": record.overtime_duration,
+					"standard_working_time": standard_working_time,
+				})
+
+	def create_overtime_details_row_for_timesheet(self, records):
+		self.overtime_details = []
+		from math import modf
+
+		standard_working_time = frappe.db.get_single_value("HR Settings", "standard_working_hours") * 3600
+		if not standard_working_time:
+			frappe.throw(_('Please Set "Standard Working Hours" in HR settings'))
+
+
+		for record in records:
+			if record.overtime_hours:
+				overtime_hours = modf(record.overtime_hours)
+				record.overtime_hours = overtime_hours[1]*3600 + overtime_hours[0]*60
+				self.append("overtime_details", {
+					"reference_document_type": "Timesheet",
+					"reference_document": record.name,
+					"date": record.overtime_on,
+					"start_date": record.start_date,
+					"end_date": record.end_date,
+					"overtime_type": record.overtime_type,
+					"overtime_duration": record.overtime_hours,
+					"standard_working_time": standard_working_time
+				})
+
+	def get_attendance_record(self):
+		if self.from_date and self.to_date:
+			records = frappe.db.sql("""SELECT overtime_duration, name, attendance_date, overtime_type, standard_working_time
+				FROM `tabAttendance`
+				WHERE
+					attendance_date >= %s AND attendance_date <= %s
+					AND employee = %s
+					AND docstatus = 1 AND status= 'Present'
+					AND (
+						overtime_duration IS NOT NULL OR overtime_duration != '00:00:00.000000'
+					)
+			""", (getdate(self.from_date), getdate(self.to_date), self.employee), as_dict=1)
```

Comment:
Use ORM for this query since it has no JOIN or SUBQUERY?

Reply:
Tried not working With ORM